lib/pages/chat_search/chat_recent_contacts_banner_widget.dart

```diff
@@ -0,0 +1,120 @@
+import 'package:collection/collection.dart';
+import 'package:dartz/dartz.dart';
+import 'package:fluffychat/app_state/failure.dart';
+import 'package:fluffychat/domain/app_state/contact/get_contacts_success.dart';
+import 'package:fluffychat/pages/chat_search/chat_recent_contacts_banner_widget_style.dart';
+import 'package:fluffychat/pages/chat_search/chat_search.dart';
+import 'package:fluffychat/presentation/model/presentation_contact.dart';
+import 'package:fluffychat/widgets/avatar/avatar.dart';
+import 'package:fluffychat/widgets/matrix.dart';
+import 'package:flutter/material.dart';
+import 'package:matrix/matrix.dart';
+
+class ChatRecentContactsBannerWidget extends StatelessWidget {
+  final ChatSearchController chatSearchController;
+  const ChatRecentContactsBannerWidget({super.key, required this.chatSearchController});
+
+  @override
+  Widget build(BuildContext context) {
+    return StreamBuilder<Either<Failure, GetContactsSuccess>>(
+      stream: chatSearchController.contactsStreamController.stream,
+      builder: (context, snapshot) {
+        if (!snapshot.hasData) {
+          return const Center(child: CircularProgressIndicator(strokeWidth: 2));
+        }
+
+        if (snapshot.hasError || snapshot.data!.isLeft()) {
+          return const SizedBox();
+        }
+
+        final contactsList = chatSearchController.fetchContactsController.getContactsFromFetchStream(snapshot.data!);
+
+        final contactsListSorted = contactsList.sorted((pre, next) => chatSearchController.comparePresentationContacts(pre, next));
+
+        if (contactsListSorted.isEmpty) {
+          return const SizedBox();
+        }
+
+        return ListView.builder(
+          shrinkWrap: true,
+          physics: const ClampingScrollPhysics(),
+          scrollDirection: Axis.horizontal,
+          itemCount: contactsListSorted.length,
+          itemBuilder: (context, index) {
+            return ChatRecentContactItemWidget(contact: contactsListSorted[index]);
+          },
+        );
+      },
+    );
+  }
+}
+
+class ChatRecentContactItemWidget extends StatelessWidget {
+  final PresentationContact contact;
+  const ChatRecentContactItemWidget({super.key, required this.contact});
+
+  @override
+  Widget build(BuildContext context) {
+    return FutureBuilder<ProfileInformation?>(
+      future: getProfile(context),
```

Comment:
are you sure, it will not have blink blink :)))

Reply:
Yes make suree